src/overlays/actors/ovl_En_Ru2/z_en_ru2.c

```diff
@@ -644,32 +820,42 @@ s32 func_80AF383C(EnRu2* this, PlayState* play) {
     return 0;
 }
 
-void func_80AF3878(EnRu2* this, PlayState* play) {
-    if (func_80AF383C(this, play) && !Play_InCsMode(play)) {
-        this->action = 16;
+/**
+ * Checks if Link is close enough to Ruto and conditionally triggers the encounter cutscene in the Water Temple.
+ */
+void EnRu2_TriggerEncounterInRange(EnRu2* this, PlayState* play) {
```

Comment:
This function spends most of its time waiting doing nothing while it waits for link, so `CheckToStartEncounter` or similar seems nice to me